src/state/actions/__test__/interests.test.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import * as rtl from '@testing-library/react';
+import configureStore from 'redux-mock-store';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import thunk from 'redux-thunk';
+import * as types from '../../types/index';
+import { getUsersInterests, getInterests } from '../interests';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const mock = new MockAdapter(axios);
+
+const interests = {
+  id: 1,
+  interest: 'AI Engineer',
+};
+
+const userInterests = {
+  id: 5,
+  interest: 'Data Science',
+  user_id: 5,
+};
+
+afterEach(rtl.cleanup);
+let wrapper;
+beforeEach(() => {
+  wrapper = rtl.render;
+});
+
+describe('Action/types interests testing', () => {
+  it('should execute get interests data', async () => {
+    const store = mockStore({});
+    const actions = store.getActions();
+
+    await store.dispatch(getInterests());
+    expect(actions[0]).toEqual({ type: types.GET_INTERESTS });
+  });
+
+  it('should execute fetch interests data with success', async () => {
+    await mock
+      .onGet(`${process.env.REACT_APP_BACKEND_URL}/interests`)
+      .reply(200, interests);
+    const expectedActions = {
+      type: types.GET_INTERESTS_SUCCESS,
+      payload: interests,
+    };
+    const store = mockStore({});
+    const actions = store.getActions();
+    await store.dispatch(getInterests(1));
+    expect(actions[1]).toEqual(expectedActions);
+  });
+  it('should execute fetch error data', async () => {
+    const code = 401;
```

Comment:
When a get request fails, it's meant to return a `404`, I'm not sure why it's returning a `401`. A `401` should be representing `unauthorized`. So it seems the `getInterests` function is only accounting for when the user doesn't have access and not accounting for when the resource fetched isn't gotten or there's an error in the request. I think this is something we can look at in the future when modifying the `getInterests` action file. That's if we end up using it. LOL